src/Platform/Action.php

```diff
@@ -0,0 +1,278 @@
+     *
+     * @return string
+     */
+    public function getDesc(): ?string
+    {
+        return $this->desc;
+    }
+
+    /**
+     * Set the value of description
+     *
+     * @param string $description
+     *
+     * @return self
+     */
+    public function desc(string $description): self
+    {
+        $this->desc = $description;
+
+        return $this;
+    }
+
+    /**
+     * Get the value of httpAliasParams
+     *
+     * @return array
+     */
+    public function getHttpAliasParams(): array
+    {
+        return $this->httpAliasParams;
+    }
+
+
+    /**
+     * Get the value of groups
+     *
+     * @return array
+     */
+    public function getGroups(): array
+    {
+        return $this->groups;
+    }
+
+    /**
+     * Set Groups
+     *
+     * @param array $groups
+     * @return self
+     */
+    public function groups(array $groups): self
+    {
+        $this->groups = $groups;
+
+        return $this;
+    }
+
+    /**
+     * Get the value of callback
+     *
+     * @return mixed
+     */
+    public function getCallback(): mixed
+    {
+        return $this->callback;
+    }
+
+    /**
+     * Set Callback
+     *
+     * @param mixed $callback
+     * @return self
+     */
+    public function callback(mixed $callback): self
+    {
+        $this->callback = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Get the value of params
+     *
+     * @return array
+     */
+    public function getParams(): array
+    {
+        return $this->params;
+    }
+
+    /**
+     * Set Param
+     *
+     * @param string $key
+     * @param mixed $default
+     * @param Validator|callable $validator
+     * @param string $description
+     * @param boolean $optional
+     * @param array $injections
+     * @return self
+     */
+    public function param(string $key, mixed $default, Validator|callable $validator, string $description = '', bool $optional = false, array $injections = []): self
+    {
+        $param = [
+            'default' => $default,
+            'validator' => $validator,
+            'description' => $description,
+            'optional' => $optional,
+            'injections' => $injections
+        ];
+        $this->httpOptions['param:' . $key] = array_merge($param, ['type' => 'param']);
+        $this->params[$key] = $param;
+
+        return $this;
+    }
+
+    /**
+     * Get the value of injections
+     *
+     * @return array
+     */
+    public function getInjections(): array
+    {
+        return $this->injections;
+    }
+
+    /**
+     * Inject
+     *
+     * @param string $injection
+     *
+     * @throws Exception
+     *
+     * @return self
+     */
+    public function inject(string $injection): self
+    {
+        if (array_key_exists($injection, $this->injections)) {
+            throw new Exception('Injection already declared for ' . $injection);
+        }
+
+        $this->httpOptions['injection:' . $injection] = [
```

Comment:
I think we should keep this concept global to all platform and strive to gradually add this feature on all libraries.